### PR TITLE
ENH: add remove_repeated_points as a GeoSeries method

### PR DESCRIPTION
**Pull request by JamesGardiner**
Implements the shapely remove_repeated_points method as part of https://github.com/geopandas/geopandas/issues/2010

### CODE REVIEW

**Comment by martinfleis** [comment]
@JamesGardiner is this ready for review?

**Comment by JamesGardiner** [comment]
@martinfleis not yet - I've been caught up with other activities the last few weeks. I'll look to get it out of draft over the weekend

**Comment by JamesGardiner** [comment]
@martinfleis I'm getting an odd failure in CI for this that I haven't had time to get to the bottom of yet. The issue is that in `Tests / ubuntu-latest, ci/envs/39-pd13-conda-forge.yaml` I get an error `TypeError: One of the arguments is of incorrect type. Please provide only Geometry objects.` - this only occurs in that specific test env and from what I can see the geometry is valid

**Comment by martinfleis** [comment]
@JamesGardiner That looks like some weird mismatch between shapely and pygeos when testing docstrings. I'll have a look.

**Comment by JamesGardiner** [comment]
Thanks @martinfleis - I did some testing and it looks like it's isolated to the Linestring in the docstring

File: geopandas/_vectorized.py [hunk 1]
```diff
@@ -722,6 +722,23 @@ def interiors(data):
...
+def remove_repeated_points(data, tolerance=0.0):
+    if compat.USE_SHAPELY_20:
+        return shapely.remove_repeated_points(data, tolerance=tolerance)
+    if compat.USE_PYGEOS and compat.SHAPELY_GE_20:
+        warnings.warn(
+            "PyGEOS does not support remove_repeated_points, and Shapely >= 2 is "
+            "installed, thus using Shapely and not PyGEOS to remove repeated points.",
+            stacklevel=4,
+        )
+        return shapely.remove_repeated_points(to_shapely(data), tolerance=tolerance)
```
[the remark this answers is not in the document]
Comment (martinfleis) [reply geopandas/_vectorized.py:734]: That makes sense. Nice catch, though I would suppose GeometryArray to complain when we give it different object in the first place...

[the remark this answers is not in the document]
Comment (martinfleis) [reply geopandas/_vectorized.py:734]: (I just can't wait to clear all this with 1.0...)

[the remark this answers is not in the document]
Comment (JamesGardiner) [reply geopandas/_vectorized.py:734]: Good catch @m-richards - I'll fix this up ASAP

**Comment by JamesGardiner** [comment]
Looks like that worked @m-richards - thanks!

**Review by martinfleis — APPROVED** [review]
Thanks!

File: geopandas/base.py [hunk 2]
```diff
@@ -859,6 +859,39 @@ def interiors(self):
         """
         return _delegate_property("interiors", self)
 
+    def remove_repeated_points(self, tolerance=0.0):
+        """Returns a ``GeoSeries`` containing a copy of the input geometry
+        with repeated points removed.
+
```
Comment (m-richards) [inline geopandas/base.py:865]:
```suggestion

        Parameters
        ----------
        tolerance : float
            radius within which a point is considered to be a repeat of another


```
Not sure how to phrase this, but think it's worth clarifying what tolerance actually is.
This is also missing from the shapely documentation, but empirically it's the radius/ distance and accounts for both coordinates:
```
shapely.remove_repeated_points(Polygon([(0, 0), (.35, .35), (0, 1), (.5, 1), (0,0)]), tolerance=.5)
Out[38]: <POLYGON ((0 0, 0 1, 0 0, 0 0))>
shapely.remove_repeated_points(Polygon([(0, 0), (.36, .36), (0, 1), (.5, 1), (0,0)]), tolerance=.5)
Out[39]: <POLYGON ((0 0, 0.36 0.36, 0 1, 0 0))>
```

**Review by m-richards — APPROVED** [review]
Thanks @JamesGardiner, very minor suggestions on docstrings.

File: geopandas/base.py [hunk 3]
```diff
@@ -899,6 +899,45 @@ def interiors(self):
...
+        ----------
+        tolerance : float
+            radius within which a point is considered to be a repeat of another
+
+
+        From the start of the coordinate sequence, each next point within the
+        tolerance is removed.
+
+        Removing repeated points with a non-zero tolerance may result in an invalid
+        geometry being returned.
```
Comment (jorisvandenbossche) [inline geopandas/base.py:916]:
```suggestion
            From the start of the coordinate sequence, each next point within the
            tolerance is removed.

            Removing repeated points with a non-zero tolerance may result in an invalid
            geometry being returned.
```

**Review by jorisvandenbossche — APPROVED** [review]
Looks good, thanks a lot! (I just pushed a small commit doing a small fix in the docstring formatting)

**Comment by martinfleis** [comment]
Thanks @JamesGardiner!

**Pull request merged**